### ISSUE TITLE
TransitionMotion defaultStyles don't reflect initial state of animation

### ISSUE THREAD

**Issue by saada** [chenglou/react-motion#416]
I explain this thoroughly here:

**Comment by richardbutler** [comment]
+1

**Comment by borela** [comment]
`defaultStyles` is used on the first interation, consider it the starting point of the overall animation. To set the initial value in `transitionMotion`, you'll need to use `willEnter`.

**Comment by saada** [comment]
I understand that's what they're intended to do but it seems like they don't function as you would expect when using TransitionMotion

**Comment by borela** [comment]
It is working http://codepen.io/borela/pen/oWqgBG `defaultStyles` set the initial value for all items that **were present when `transitionMotion` was mounted**, `defaultStyles` is only used once, after that, any new item, get its initial value from `willEnter` and `defaultStyles` is ignored.

In the article, you are constantly generating new items, and all those items are getting their initial values from `willEnter`(which is not defined). But I agree that the documentation is a bit confusing on this one.

**Comment by saada** [comment]
The problem is that each item needed to start at a different position... Will enter assumes that all items start at the same position, no?

**Comment by borela** [comment]
No, you can specify a different position for each item too because `willEnter` receives the new item as an argument, you already saving the particle data, just need some adjustments:

```js
styles = particles.map(particle => ({
    key: particle.key,
    // You are saving the particle data here, so we can access it in the willEnter.
    data: particle,  
    style: {
      // I am setting the final position and opacity here.
      translateX: spring(particle.data.x2, particleSpring),
      translateY: spring(particle.data.y2, particleSpring),
      opacity: spring(1, particleSpring)
    }
}))

// You can access the particle data.
willEnter(particle) {
 // The structure particle argument is:
 // particle = {
 //   key: '...',
 //   data: '...', The particle data you saved.
 //   style: {
 //   The final style set in "style" before.
 // }

 // Returns the starting position and opacity.
  return {
      translateX: particle.data.x1,
      translateY: particle.data.y1,
      opacity: .5
  }
}
```

**Comment by richardbutler** [comment]
Thanks a lot for the clarification, that does make much more sense now.
